packages/prop-house-communities/src/strategies/balanceOfErc721Multiple.ts

```diff
@@ -0,0 +1,26 @@
+import { Strategy } from '../types/Strategy';
+import BalanceOfABI from '../abi/BalanceOfABI.json';
+import { Contract } from 'ethers';
+import BigNumber from 'bignumber.js';
+import { parseBlockTag } from '../utils/parseBlockTag';
+import { Provider } from '@ethersproject/providers';
+
+/**
+ * Calculates `balanceOf` for multiple ERC-721 contracts
+ */
+export const balanceOfErc721Multiple = (addresses: string[], multiplier: number = 1): Strategy => {
+  return async (
+    userAddress: string,
+    communityAddress: string,
+    blockTag: number,
+    provider: Provider,
+  ) => {
+    let sum = new BigNumber(0);
+    for (let i = 0; i < addresses.length; i++) {
+      const contract = new Contract(addresses[i], BalanceOfABI, provider);
+      const balance = await contract.balanceOf(userAddress, { blockTag: parseBlockTag(blockTag) });
+      sum = sum.plus(new BigNumber(balance.toString()));
+    }
+    return sum.times(multiplier).toNumber();
+  };
```

Comment:
Currently, these will execute sequentially, waiting for the previous call to resolve before making the next. Instead, use `map` and `Promise.all` to make these calls in parallel.

e.g.
```ts
export const balanceOfErc721Multiple = (addresses: string[], multiplier: number = 1): Strategy => {
  return async (
    userAddress: string,
    _communityAddress: string,
    blockTag: number,
    provider: Provider,
  ) => {
    const balances = await Promise.all(addresses.map(async address => {
      const contract = new Contract(address, BalanceOfABI, provider);
      const balance = await contract.balanceOf(userAddress, { blockTag: parseBlockTag(blockTag) });
      return new BigNumber(balance.toString());
    }));
    return BigNumber.sum(...balances).times(multiplier).toNumber();
  };
};
```

Reply:
🧠  ty